src/Illuminate/Container/Util.php

```diff
@@ -84,4 +89,58 @@ public static function getContextualAttributeFromDependency($dependency)
     {
         return $dependency->getAttributes(ContextualAttribute::class, ReflectionAttribute::IS_INSTANCEOF)[0] ?? null;
     }
+
+    /**
+     * This has been extracted from API Platform
+     *
+     * @param string $directory
+     *
+     * @return array<class-string, \ReflectionClass>
+     */
+    public static function getReflectionClassesFromDirectory(string $directory): array
+    {
+        $id = hash('xxh3', $directory);
+        if (isset(self::$localCache[$id])) {
+            return self::$localCache[$id];
+        }
+
+        $includedFiles = [];
+        $iterator = new \RegexIterator(
+            new \RecursiveIteratorIterator(
+                new \RecursiveDirectoryIterator($directory, \FilesystemIterator::SKIP_DOTS | \FilesystemIterator::FOLLOW_SYMLINKS),
+                \RecursiveIteratorIterator::LEAVES_ONLY
+            ),
+            '/^.+\.php$/i',
+            \RecursiveRegexIterator::GET_MATCH
+        );
+
+        foreach ($iterator as $file) {
+            $sourceFile = $file[0];
+
+            if (!preg_match('(^phar:)i', (string) $sourceFile)) {
+                $sourceFile = realpath($sourceFile);
+            }
+
+            try {
+                require_once $sourceFile;
```

Comment:
This is definitely going to be an issue if the discovered file is a blade template or a pest unit file. Still looking to improve that...